src/site/content/en/blog/commonjs-larger-bundles/index.md

```diff
@@ -0,0 +1,244 @@
+---
+title: How CommonJS is making your bundles larger
+subhead: Learn how CommonJS modules are impacting the tree-shaking of your application
+authors:
+  - mgechev
+date: 2020-05-07
+updated: 2020-05-07
+hero: hero.jpg
+alt: How CommonJS is making your bundles larger
+description:
+  CommonJS modules are very dynamic, which prevents JavaScript
+  optimizers and bundles perform advanced optimizations over them.
+tags:
+  - blog
+  - javascript
+  - modules
+  - commonjs
+  - tree-shaking
+---
+
+In this post, we'll look into what CommonJS is and why it's making your JavaScript bundles larger than necessary.
+
+Summary: **To ensure the bundler can successfully optimize your application, avoid depending on CommonJS modules, and use ES2015 module syntax in your entire application.**
+
+
+## What's CommonJS?
+
+CommonJS is a standard from 2009 that established conventions forJavaScript modules. It was initially intended for use outside of the web browser, primarily for server-side applications.
+
+With CommonJS you can define modules, export functionality from them, and import them in other modules. For example, the snippet below defines a module which exports five functions: `add`, `subtract`, `multiply`, `divide`, and `max`:
+
+```javascript
+// utils.js
+const { maxBy } = require('lodash-es');
+const fns = {
+  add: (a, b) => a + b,
+  subtract: (a, b) => a - b,
+  multiply: (a, b) => a * b,
+  divide: (a, b) => a / b,
+  max: arr => maxBy(arr)
+};
+
+Object.keys(fns).forEach(fnName => module.exports[fnName] = fns[fnName]);
+```
+
+Later on, another module can import and use some or all of these functions:
+
+```javascript
+// index.js
+const { add } = require(‘./utils');
+console.log(add(1, 2));
+```
+
+Invoking `index.js` with `node` will output the number `3` in the console.
+
+Because of the lack of a standardized module system in the browser, in the early 2010s, CommonJS became a popular module format for JavaScript client-side libraries as well.
```

Comment:
```suggestion
Because of the lack of a standardized module system in the browser in the early 2010s, CommonJS became a popular module format for JavaScript client-side libraries as well.
```